File: packages/frame_ble/examples/stay_awake_true.py
import asyncio
from frame_ble import FrameBle

async def main():
    frame = FrameBle()

    try:
        await frame.connect()

        # stop any application, if running, so we can send lua commands
        await frame.send_break_signal()

        # Keep Frame awake even in charging cradle (for development)
        await frame.send_lua("frame.stay_awake(true);print(0)", await_print=True)
        print("Frame will stay awake - even in the charging cradle - until frame.send_lua('frame.stay_awake(false)')")

        await frame.disconnect()

    except Exception as e:
        print(f"Not connected to Frame: {e}")
        return

if __name__ == "__main__":
    asyncio.run(main())

File: webbluetooth/packages/brilliant-ble/example/stay-awake-true.js
import { FrameBle } from 'frame-ble';

export async function run() {
  const frameBle = new FrameBle();

  // Connect to Frame
  const deviceId = await frameBle.connect();

  // Send a break signal to Frame in case it is in a loop/main.lua
  await frameBle.sendBreakSignal();

  // Keep Frame awake even in charging cradle (for development)
  await frameBle.sendLua("frame.stay_awake(true);print(0)", {awaitPrint: true})

  // Disconnect from Frame
  await frameBle.disconnect();
};
